xmitgcm/test/test_mds_store.py

```
@@ -195,6 +196,44 @@ def test_open_dataset_no_meta(all_mds_datadirs):
                                          nz=nz, **kwargs)
 
 
+def test_open_dataset_2D_diags(all_mds_datadirs):
+    # convert 3D fields with only 2D diagnostic output
+    # https://github.com/xgcm/xmitgcm/issues/140
+    dirname, expected = all_mds_datadirs
+
+    shape = expected['shape']
+
+    nz = shape[0]
+    ny, nx = shape[-2:]
+    shape_2d = shape[1:]
+    dims_2d = ('j', 'i')
+    if expected['geometry']=='llc':
+        dims_2d = ('face',) + dims_2d
+        ny = nx*shape[-3]
+    dims_3d = dims_2d if nz==1 else ('k',) + dims_2d
+    dims_2d = ('time',) + dims_2d
+    dims_3d = ('time',) + dims_3d
+
+    it = expected['test_iternum']
+    kwargs = dict(iters=it, geometry=expected['geometry'], read_grid=False,
+                  swap_dims=False)
+
+    to_hide = ['T.%010d.meta' % it, 'T.%010d.data' % it]
+    with hide_file(dirname, *to_hide):
+
+        ldir = py.path.local(dirname)
+        old_prefix = 'Eta.%010d' % it
+        new_prefix = 'T.%010d' % it
+        for suffix in ['.data', '.meta']:
+            lp = ldir.join(old_prefix + suffix)
+            lp.copy(ldir.join(new_prefix + suffix))
+
+        # currently raises ONLY with LLC-type datasets, but not regular grids
+        # WHY?!?
+        ds = xmitgcm.open_mdsdataset(dirname, prefix=['T'], **kwargs)
```

My question for the group, including @ifenty, is, what sort of API do we want here? Do we want it to "just work" when these 2D diags are encountered, or do we want to raise an error by default? In this case, we would have a new keyword argument like `allow_2d_diags=True` which would turn on the new feature of correctly parsing 2D slices of 3D diags.

---

My opinion is to require the keyword `allow_2d_diags`. Along with this, we can use the if statements above to print out a useful error message. Something like 

```
#Expect 3 dims, vertical and 2 horizontal 
ndims_expected = 3 
if self.llc:
    # llc geometry adds face dim 
    ndims_expected += 1 
  
if len(dims) == 3 and data.ndim == ndims_expected-1:
    if allow_2d_diags: 
        dims = dims[1:] 
        vname = '%s_2D' % vname 
    else: 
        raise TypeError('Variable %s from file prefix %s has unexpected number of dimensions. \n'
                        'Set allow_2d_diags=True to read 2D diagnostic of a typically 3D variable' % (vname,prefix))
```
Perhaps a different error should be used, but this is my general idea. 

I'd also be happy to turn the notebook I've been using for testing into documentation.